net/bridge.go

```diff
@@ -292,9 +292,7 @@ func EnsureBridge(procPath string, config *BridgeConfig, log *logrus.Logger, ips
 		return bridgeType, errors.Wrapf(err, "configuring ARP cache on bridge %q", config.WeaveBridgeName)
 	}
 
-	// NB: No concurrent call to Expose is possible, as EnsureBridge is called
-	// before any service has been started.
```

Comment:
This is a bit of a question.

But `Expose` can already by hit concurrently as its an http endpoint too?

There are some comments in the `Expose` helpers like
> [// The IP addr might have been already set by a concurrent request, just ignore then](https://github.com/weaveworks/weave/blob/master/net/expose.go#L61)

Which suggest it should be okay? We probably don't need to wrap Expose in a Lock or anything?

Reply:
Your point seems valid.  Maybe the thinking was that an address could not be added while we were traversing the list.
Maybe it relates to some earlier state of the code where concurrency was an issue.